dvc: replace pkg concept with external repo

efiop opened the pull request:
`dvc import` is now renamed to `dvc import-url`. `dvc pkg import` renamed into `dvc import` and `dvc pkg get` into `dvc get`. Also, for symmetry, introduced `dvc get-url`. Along with some other minor modifications. The workflow now looks like:

```
# downloads and creates dvc file
dvc import https://github.com/iterative/dvc path/data --rev 0.41.3

# only downloads data
dvc get https://github.com/iterative/dvc path/data --rev 0.41.3
```
Signed-off-by: Ruslan Kuprieiev <ruslan@iterative.ai>

* [ ] Have you followed the guidelines in our
      [Contributing document](https://dvc.org/doc/user-guide/contributing)?

* [ ] Does your PR affect documented changes or does it add new functionality
      that should be documented? If yes, have you created a PR for
      [dvc.org](https://github.com/iterative/dvc.org) documenting it or at
      least opened an issue for it? If so, please add a link to it.

-----

shcheklein commented on dvc/external_repo.py:
```diff
@@ -0,0 +1,153 @@
+            return
+
+        try:
+            os.makedirs(self.repos_dir)
```
why not compat makedirs?

shcheklein commented on dvc/external_repo.py:
```diff
@@ -0,0 +1,153 @@
+        #
+        # Note that tempfile.TemporaryDirectory is using symlinks to tmpfs, so
+        # we won't be able to use move properly.
+        tmp_dir = os.path.join(self.repos_dir, "." + str(uuid.uuid4()))
```
is it possible to use pathlib?

efiop replied:
Not really, our Repo and other stuff are not meant to be used with it.

shcheklein commented on dvc/external_repo.py:
```diff
@@ -0,0 +1,153 @@
+                shutil.rmtree(tmp_dir)
+            raise
+
+        shutil.move(tmp_dir, self.path)
```
is it atomic for directories on different partitions?

efiop replied:
tmp_dir is created beside self.path, so they are on the same partition.

shcheklein replied:
yep, I see. I just misinterpreted the comment `Note that tempfile.TemporaryDirectory is using symlinks to tmpfs, so
        # we won't be able to use move properly.`

shcheklein commented on dvc/external_repo.py:
```diff
@@ -0,0 +1,153 @@
+            cache_config.set_dir(cache_dir, level=Config.LEVEL_LOCAL)
+            repo.scm.git.close()
+
+        if self.installed:
```
should be probably part of the `def install`

shcheklein commented on dvc/command/get_url.py:
```diff
@@ -0,0 +1,39 @@
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    get_parser.add_argument(
+        "url", help="See `dvc import-url -h` for full list of supported URLs."
```
import-url -> get-url

shcheklein replied:
I see, it's intentional, right?

efiop replied:
Yes.

shcheklein commented on dvc/command/get_url.py:
```diff
@@ -0,0 +1,39 @@
+        "url", help="See `dvc import-url -h` for full list of supported URLs."
+    )
+    get_parser.add_argument(
+        "out", nargs="?", help="Destination path to put data to."
```
do we want to support a short option `-o`?

efiop replied:
`-o` is mostly needed for importing repo, so there is no real need in it for `import-url` or `get-url`.

shcheklein commented on dvc/repo/get_url.py:
```diff
@@ -0,0 +1,19 @@
+        src = os.path.abspath(url)
+        out = os.path.abspath(out)
+
+    dep, = dependency.loads_from(None, [src])
```
will it support remote:// protocol? it needs to read a global config for that, right?

efiop replied:
It won't, at least for now. `get-url` doesn't care about Repo's at all. I would keep it this way untill someone asks for this feature.

shcheklein commented on dvc/repo/get_url.py:
```diff
@@ -0,0 +1,19 @@
+    out = out or os.path.basename(urlparse(url).path)
+
+    if os.path.exists(url):
+        src = os.path.abspath(url)
```
src is undefined if `not os.path.exists(url)` and is referenced below

mroutis commented:
what about backwards compatibility, @efiop ? `dvc import` -> `dvc import-url`, if this is already released, should we let know the users somehow?

efiop commented:
@mroutis We do let them know with a warning in dvc/command/imp.py.

mroutis commented:
True, @efiop :+1:

guysmoilov commented:
Hey guys, I just tripped all over this when running our tutorial at https://dagshub.com/docs/pipeline/
in preparation for the workshop on DVC we're planning to do on Sunday.
I guess I'm out of date here, but why rename `import` to `import-url` instead of giving the new command the new name?
Also, the docs in the main site are out of date: https://dvc.org/doc/commands-reference/import

guysmoilov commented:
I would also consider people who may have written shell scripts with `dvc import`, a warning message would not really be relaxing for them...

efiop commented:
@guysmoilov Thanks for the feedback! We've decided to rename it so that a more powerful repo-import feature would have a better name(we've also introduced new `dvc get` command for downloading from repos, and it felt weird having it in pair with `repo-import` or `import-repo`, so we've decided to sacrifice `import-url`). Sorry for the inconvenience. We've added a warning and bumped the version to something significant(not a major version though :slightly_frowning_face:) to help somewhat mitigate that, but it was clearly not enough.

mroutis commented:
Agree, @guysmoilov ! @efiop , maybe writing something in the #announcements channel on Discord?

mroutis commented:
@shcheklein just did it :sweat_smile:

shcheklein commented:
@mroutis @efiop I have posted an announcement. @guysmoilov sorry for the confusion and inconvenience :( It was not an easy decision.I really hope that you will try the new set of commands that are in a preview mode - `import` and `get`. I think they alone can create lot of value to users and enable really great use cases around them.

As for the documentation update - it's in progress. Import will be renamed in a few hours.

jorgeorpinel commented:
Probably too late but I agree with Guy. Reusing a previous command name for a similar feature as before seems really confusing and unnecessary. There's an endless number of other words that could've been used. Anyway, probably not the end of the world and I'm sure there was a deep discussion about it so, moving on! Working on the documentation updates in PR #464
